mopidy_musicbox_webclient/static/js/functionsvars.js

```diff
@@ -176,185 +176,156 @@ function albumTracksToTable (pl, target, uri) {
+    return !track.album || !track.album.name || !previousTrack || !previousTrack.album || !previousTrack.album.name ||
+           (previousTrack && previousTrack.album && previousTrack.album.name && track.album.name !== previousTrack.album.name)
+}
+
+function isMultiTrackAlbum (track, nextTrack) {
+    // 'true' if there are more tracks of the same album after this one.
+    return nextTrack.album && nextTrack.album.name && track.album && track.album.name && track.album.name === nextTrack.album.name
+}
+
+function validateTrackName (track) {
+    // Create name if there is none
+    var name = ''
+    if (!track.name || track.name === '') {
+        name = track.uri.split('/')
+        name = decodeURI(name[name.length - 1]) || 'Track ' + String(i)
     } else {
-        name = song.name
+        name = track.name
     }
-    songLi = '<li class="song albumli" id="' + liID + '" tlid="' + tlid + '">' +
-             '<a href="#" class="moreBtn" onclick="return popupTracks(event, \'' + uri + '\',\'' + song.uri + '\',\'' + tlid + '\');">' +
-             '<i class="fa fa-ellipsis-v"></i></a>' +
-             '<a href="#" onclick="return playTrackQueueByTlid(\'' + song.uri + '\',\'' + tlid + '\');">' +
-             '<h1 class="trackname">' + name + '</h1></a>' +
-             '</li>'
-    return songLi
+    return name
 }
 
 function resultsToTables (results, target, uri) {
     if (!results) {
         return
     }
-    var tlids = []
-    if (target === CURRENT_PLAYLIST_TABLE) {
-        for (i = 0; i < results.length; i++) {
-            tlids[i] = results[i].tlid
-            results[i] = results[i].track
-        }
-    }
+    $(target).html('')
+    $(target).attr('data', uri)
 
-    var newalbum = []
-    var newtlids = []
-    // keep a list of track URIs for retrieving of covers
+    var track, previousTrack, nextTrack, tlid
+    var albumTrackSeen = 0
+    var renderAlbumInfo = true
+    var liID = ''
+    // Keep a list of track URIs for retrieving of covers
     var coversList = []
-    var nextname = ''
-    var count = 0
-    $(target).html('')
 
-    // break into albums and put in tables
     var html = ''
-    var tableid, j, artistname, alburi, name, iconClass
+    var tableid, artistname, name, iconClass
     var targetmin = target.substr(1)
-    $(target).attr('data', uri)
     var length = 0 || results.length
+
+    // Break into albums and put in tables
     for (i = 0; i < length; i++) {
-        // create album if none extists
-        if (!results[i].album) {
-            results[i].album = {'__model__': 'Album'}
+        previousTrack = track
+        track = results[i]
+        tlid = ''
+        if (i < length - 1) {
+            nextTrack = results[i + 1]
         }
-        // create album uri if there is none
-        if (!results[i].album.uri) {
-            results[i].album.uri = 'x'
+        if (track.__model__ === 'TlTrack') {
```

Comment:
Is looking at the `__model__` property better than looking for the existence of a `tlid` property? It seems more brittle.